keras/src/callbacks/orbax_checkpoint_test.py

```diff
@@ -577,3 +587,801 @@ def compare_nested_dicts(orig_dict, loaded_dict):
+            for orig, loaded in zip(original_weights, loaded_weights)
+        )
+        self.assertTrue(weights_changed, "Weights should change after loading")
+
+    @pytest.mark.skipif(
+        backend.backend() != "jax", reason="Sharding tests require JAX backend"
+    )
+    def test_load_checkpoint_resharding_jax(self):
+        """Test load_checkpoint works with distribution set (JAX only)."""
+        import os
+
+        import jax
+
+        from keras.src.distribution import DeviceMesh
+        from keras.src.distribution import LayoutMap
+        from keras.src.distribution import ModelParallel
+        from keras.src.distribution import TensorLayout
+        from keras.src.distribution import set_distribution
+
+        # Check if we have at least 1 device
+        devices = jax.devices()
+        if len(devices) < 1:
+            self.skipTest("Test requires at least 1 JAX device")
+
+        # Skip test if there are more than 2 devices, as these tests are
+        # designed for 2-device scenarios and may not work with more devices
+        if len(devices) > 2:
+            self.skipTest(f"Test for 2 devices, but {len(devices)} available")
+
+        num_devices = min(2, len(devices))
+
+        # Configure JAX to use virtual devices if needed
+        original_xla_flags = os.environ.get("XLA_FLAGS", "")
+        if num_devices < 2:
+            os.environ["XLA_FLAGS"] = "--xla_force_host_platform_device_count=2"
+            # Re-check devices after setting flag
+            devices = jax.devices()
+            num_devices = min(2, len(devices))
+
+        try:
+            print(f"Available devices: {devices}, using {num_devices} devices")
+
+            # Set up distribution based on available devices
+            if num_devices >= 2:
+                # Multi-device distribution
+                device_mesh = DeviceMesh((num_devices,), axis_names=["data"])
+                layout_map = LayoutMap(device_mesh)
+                layout_map["dense_layer/kernel"] = TensorLayout(
+                    axes=("data", None)
+                )
+                layout_map["dense_layer/bias"] = TensorLayout(axes=(None,))
+                layout_map["output_layer/kernel"] = TensorLayout(
+                    axes=(None, "data")
+                )
+                layout_map["output_layer/bias"] = TensorLayout(axes=(None,))
+            else:
+                # Single device distribution
+                device_mesh = DeviceMesh((1,), axis_names=["data"])
+                layout_map = LayoutMap(device_mesh)
+                layout_map["dense_layer/kernel"] = TensorLayout(
+                    axes=(None, None)
+                )
+                layout_map["dense_layer/bias"] = TensorLayout(axes=(None,))
+                layout_map["output_layer/kernel"] = TensorLayout(
+                    axes=(None, None)
+                )
+                layout_map["output_layer/bias"] = TensorLayout(axes=(None,))
```

Comment:
But this is not proving anything if there is only 1 device.

Reply:
fixed the issue by adding a skip condition for single-device scenarios.